frontend/src/views/Files.vue

```diff
@@ -72,7 +69,26 @@ export default {
     this.fetchData();
   },
   watch: {
-    $route: "fetchData",
+    $route: function (to, from) {
+      if (from.path.endsWith("/")) { 
+        if (to.path.endsWith("/"))   {
+          window.sessionStorage.setItem('listFrozen', "false");
+          this.fetchData();
+          return;
+        } else {
+          window.sessionStorage.setItem('listFrozen', "true");
+          this.fetchData();
+          return;        
+        }
+      } else if (to.path.endsWith("/")) {
+        this.$store.commit("updateRequest", {});
+        this.fetchData();
+        return;
+      } else {
+        this.fetchData();
+        return;
+      } 
+    }, 
```

Comment:
here we judge if the routing is from a dir (listing) to a file (preview), a dir to another dir, a file to a file, or a file to a dir. we only allow the listing to reload when it's from a dir to a dir.